js/src/run_trees.ts

```diff
@@ -489,12 +516,23 @@ export class RunTree implements BaseRun {
     try {
       const runtimeEnv = getRuntimeEnvironment();
       if (this.replicas && this.replicas.length > 0) {
-        for (const [projectName] of this.replicas) {
-          const runCreate = this._remapForProject(
-            projectName,
-            runtimeEnv,
-            true
-          );
+        for (const replica of this.replicas) {
+          let runCreate;
+          if (!replica.projectName) {
+            runCreate = this._convertToCreate(this, runtimeEnv, true);
+          } else {
+            runCreate = this._remapForProject(
+              replica.projectName,
+              runtimeEnv,
+              true
+            );
+          }
+          if (replica.apiKey) {
+            runCreate.apiKey = replica.apiKey;
+          }
+          if (replica.apiUrl) {
+            runCreate.apiUrl = replica.apiUrl;
```

Comment:
Would prefer to pass these separately into `createRun` - or hot take maybe even give each replica its own client?

Reply:
hmmm would that be worse

Reply:
Yeah probably too hot of a take